File: backend/src/main/java/org/example/backend/domain/tradecomment/dto/TradeCommentDeleteRequestDto.java
package org.example.backend.domain.tradecomment.dto;

import org.example.backend.domain.trade.enums.BoardType;

public record TradeCommentDeleteRequestDto(
    BoardType boardType,
    Long tradeId,
    Long commentId,
    Long memberId
) {
    public static TradeCommentDeleteRequestDto of(
        BoardType boardType,
        Long tradeId,
        Long commentId,
        Long memberId
    ) {
        return new TradeCommentDeleteRequestDto(boardType, tradeId, commentId, memberId);
    }
}

File: backend/src/main/java/org/example/backend/domain/tradecomment/dto/TradeCommentDeleteRequestDto.kt
package org.example.backend.domain.tradecomment.dto

import org.example.backend.domain.trade.enums.BoardType

data class TradeCommentDeleteRequestDto(
    val boardType: BoardType,
    val tradeId: Long,
    val commentId: Long,
    val memberId: Long
) {
    companion object {
        fun of(
            boardType: BoardType,
            tradeId: Long,
            commentId: Long,
            memberId: Long
        ): TradeCommentDeleteRequestDto {
            return TradeCommentDeleteRequestDto(
                boardType = boardType,
                tradeId = tradeId,
                commentId = commentId,
                memberId = memberId
            )
        }
    }
}
